### ISSUE TITLE
Can not build on OS X

### ISSUE THREAD

**Issue by Luthaf** [scafacos/scafacos#19]
Trying to build this library on OS X 10.9.5, with gcc-4.9, gfortran-4.9 and mpich gives me the following error:

```
Making all in common/pfft
make[3]: *** No rule to make target `all'.  Stop.
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```
Is this an issue in the initial steps: 
```
./bootstrap
./configure
```

**Comment by mpip** [comment]
Please send me the output of bootstrap, configure and make. Since this can be very long save the output to a logfile, i.e., call
`./bootstrap 2>&1 | tee build.log`
`./configure $YOURFLAGS  2>&1 | tee -a build.log`
`make 2>&1 | tee -a build.log`
and send the file `build.log` directly to my email.

**Comment by Luthaf** [comment]
Here are the logs : https://gist.github.com/Luthaf/08371122858b6ff5dd8a

I tried to build ScaFaCoS on CentOs, and I coud not pass the bootstrap step. Here is the log : https://gist.github.com/Luthaf/1eddb7c5e9e50024ce08

I also ran into another error while running `make clean`:
```
Making clean in common/pfft
make[2]: *** No rule to make target `clean'.  Stop.
make[1]: *** [clean-recursive] Error 1
make: *** [clean-recursive] Error 1
```

**Comment by mh0fm4** [comment]
Am 27.01.2015 um 00:37 schrieb Luthaf:
> Trying to build this library on OS X 10.9.5, with gcc-4.9, gfortran-4.9
> and mpich gives me the following error:
> 
> |Making all in common/pfft
> make[3]: *** No rule to make target `all'.  Stop.
> make[2]: *** [all-recursive] Error 1
> make[1]: *** [all-recursive] Error 1
> make: *** [all] Error 2
> |

Are the directories common/pfft and common/pnfft empty?

These are submodules located in other repositories, so you have to do

  git submodule update --init --recursive

(within your cloned scafacos directory) to fetch them.

Afterward, please continue with bootstrap.

**Comment by Luthaf** [comment]
The submodule init has done its job ! Shouldn't some lines about how to build be added to the README ? 

Still, I have another build error, in the Fortran part of the code : https://gist.github.com/Luthaf/08371122858b6ff5dd8a#file-fortran-build-log
Any ideas of when it's may come from ?

**Comment by mh0fm4** [comment]
Yes, giving some more build information would be good.

The build error is a known problem. The Fortran code of the FMM solver has some "issues" that prevent a successful build with gfortran version 4.7 or later where the "-fno-whole-file" option was removed. Unfortunately, the only solutions are to use an older version of gfortran or to disable the build of the FMM solver completely (with --disable-fcs-fmm).

**Comment by Luthaf** [comment]
Ok, I am making some progress. I was able to build on CentOS, but OS X is still failing.

The next error I have is 
```
Making all in pepc
/Library/Developer/CommandLineTools/usr/bin/make  all-recursive
Making all in src
/bin/sh ../libtool  --tag=FC   --mode=compile mpif90 -DHAVE_FCONFIG_H -I. -I..    -I../../.. -g -O2 -ffree-form -ffree-line-length-none -fopenmp -c -o module_coulomb_kernels.pp.lo -cpp module_coulomb_kernels.pp.f90
libtool: compile:  mpif90 -DHAVE_FCONFIG_H -I. -I.. -I../../.. -g -O2 -ffree-form -ffree-line-length-none -fopenmp -c -cpp module_coulomb_kernels.pp.f90 -o module_coulomb_kernels.pp.o
Building file 'module_debug.pp.f90'
cat ../fconfig.h ./pepc_debug.h module_debug.f90 | \
	mpicc -E -x c -C -P -P -C -DSVNREVISION="\"\"" -DSL_PEPC_PREFIX=fcs_ -DFFLAGS="\"-I../../..\"" -DCFLAGS="\"\"" -DLDFLAGS="\"\"" -DMACH="\"\"" -DCOMPILER="\"??\"" -DWALKALGORITHM="\"walk_pthreads\""  -I. -I.. - | \
	/usr/bin/sed -e '/\/\*/{:l;s,\(.*\)/\*\(.*\)\*/,\1!\2,;t l}' - | \
	/usr/bin/sed -e '/\/\*/,/.*\*\//{s,\(.*\)\*/,!\1,; t e; s,/\*\(.*\),!\1,; t e; s,^,!,; :e;}' - | \
	/usr/bin/sed -e '/\/\*/,/.*\*\//{s,\(.*\)\*/,!\1,; t e; s,/\*\(.*\),!\1,; t e; s,^,!,; :e;}' - > module_debug.pp.f90
sed: 2: "/\/\*/{:l;s,\(.*\)/\*\( ...": unexpected EOF (pending }'s)
sed: 2: "/\/\*/,/.*\*\//{s,\(.*\ ...": unexpected EOF (pending }'s)
sed: 2: "/\/\*/,/.*\*\//{s,\(.*\ ...": unexpected EOF (pending }'s)
<stdin>:5:21: warning: missing terminating ' character [-Winvalid-pp-token]
! The size of `int *', as computed by sizeof.
                    ^
clang: error: unable to execute command: Broken pipe: 13
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple LLVM version 6.0 (clang-600.0.56) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix
clang: note: diagnostic msg: PLEASE submit a bug report to http://developer.apple.com/bugreporter/ and include the crash backtrace, preprocessed source, and associated run script.
clang: note: diagnostic msg: Error generating preprocessed source(s) - ignoring input from stdin.
clang: note: diagnostic msg: Error generating preprocessed source(s) - no preprocessable inputs.
make[5]: *** [module_debug.pp.f90] Error 1
make[4]: *** [all-recursive] Error 1
make[3]: *** [all] Error 2
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

Its looks like a difference between the gnu `sed` and the BSD `sed` (the latter being the OS X version).

**Comment by mh0fm4** [comment]
Thanks for keeping on trying. I could verify the problems with BSD sed and have commited a fix.

**Comment by mh0fm4** [comment]
Fixed with commit 50dec9482ec5b1968990aa99fefe00e9aff42f76.

**Comment by Luthaf** [comment]
Nice, thanks !
